pwiz_tools/Skyline/TestUtil/TestFunctional.cs

```diff
@@ -2204,32 +2204,25 @@ public static void PasteSmallMoleculeListNoAutoManage()
 
         // Importing a small molecule transition list typically provokes a dialog asking whether or not to automatically manage the resulting transitions
         // The majority of our tests were written before this was an option, so we dismiss the dialog by default and the new nodes are automanage OFF
-        public static void DismissAutoManageDialog(SrmDocument docCurrent, bool? expectAutoManage = null)
+        public static void DismissAutoManageDialog()
         {
-            MultiButtonMsgDlg wantAutoManageDlg;
-            if (expectAutoManage ?? false)
+            var autoManageDlg = WaitForOpenForm<MultiButtonMsgDlg>();
+            if (autoManageDlg == null)
             {
-                wantAutoManageDlg = WaitForOpenForm<MultiButtonMsgDlg>();
+                AssertEx.Fail($@"Expected dialog with message like ""${Resources.SkylineWindow_ImportMassList_Do_you_want_to_use_the_document_settings_to_automanage_these_new_transitions}""");
             }
-            else
+            // Make sure we haven't intercepted some other dialog
+            else if (IsFormattedMessage(Resources.SkylineWindow_ImportMassList_Do_you_want_to_use_the_document_settings_to_automanage_these_new_transitions, autoManageDlg.Message))
```

Comment:
This feels like it should be replaced with a simple assert, getting rid of the if clause entirely and the need to call AssertEx.Fail() yourself.